### PR TITLE
Test -redhat-x version parsing (NCL-1915).

### PR DESCRIPTION
**Pull request by matejonnet**

### CODE REVIEW

**Comment by thescouser89** [comment]
retest this please

**Comment by thescouser89** [comment]
add to whitelist

**Comment by thescouser89** [comment]
retest this please

**Comment by thescouser89** [comment]
retest this pleaes

**Comment by thescouser89** [comment]
retest this please

**Comment by jbartece** [comment]
add to whitelist

**Comment by jbartece** [comment]
retest this please

**Comment by janinko** [comment]
Is this meant to solve anything or is it just a test? What is it testing?

**Comment by janinko** [comment]
retest this please

**Comment by thescouser89** [comment]
retest this please

**Comment by thescouser89** [comment]
Formatting errors :)

**Comment by matejonnet** [comment]
No fixes only test to make sure the correct version is returned.
Do you have code-style settings to import to IDE ?

**Comment by thescouser89** [comment]
@matejonnet There's this: https://github.com/project-ncl/dependency-analysis/blob/master/eclipse-codeStyle.xml

**Comment by matejonnet** [comment]
Auto-formatting with added code style doesn't match previous code style.

File: reports-backend/src/main/java/org/jboss/da/reports/backend/impl/VersionFinderImpl.java [hunk 1]
```diff
@@ -26,15 +26,23 @@
...
     Logger log;
 
-    @Inject
     private AproxConnector aproxConnector;
 
-    @Inject
     private VersionParser osgiParser;
 
+    @Deprecated //CDI constructor injection workaround
+    public VersionFinderImpl() {
```
Comment (janinko) [inline reports-backend/src/main/java/org/jboss/da/reports/backend/impl/VersionFinderImpl.java]: Isn't better to have the constructor with protected or package visibility?
Or you can use Mockito in the test for injecting (using ``@Spy`` and ``@InjectMocks`` annotations). See VersionFinderTest.

**Comment by janinko** [comment]
Actually, it might be better to add the test to already existing ``org.jboss.da.reports.backend.impl.VersionFinderTest`` (different package name).

**Comment by janinko** [comment]
retest this please

**Comment by janinko** [comment]
There is wrong formatting. Running this should fix it:
```
$ mvn formatter:format -DtestsuiteContainer
```

**Pull request merged**